lib/ansible/plugins/doc_fragments/constructed.py

```diff
@@ -29,6 +29,36 @@ class ModuleDocFragment(object):
     description: Add hosts to group based on the values of a variable.
     type: list
     default: []
+    elements: dict
+    suboptions:
+      parent_group:
+        type: str
+        description: parent group for keyed group
+      prefix:
+        type: str
+        description: A keyed group name will start with this prefix
+        default: ''
+      separator:
+        type: str
+        description: separator used to build the keyed group name
+        default: "_"
+      key:
+        type: str
+        description:
+        - The key from input dictionary used to generate groups
+      default_value:
+        description:
+        - The default name when the key's value is empty or null.
```

Comment:
Just to clarify what I found unclear, so you can update it in your own words - I thought `name` was unclear because it's only a portion of the group name - the prefix and separator are still used. And I interpret `null` it in the programming sense of the word - in YAML null is `~` and in Python null is `None`.